File: ApemanPlugin/src/main/kotlin/apeman_core/features_extraction/calculators/method/NumTernaryMethodCalculator.java
package apeman_core.features_extraction.calculators.method;

import com.intellij.psi.JavaRecursiveElementVisitor;
import com.intellij.psi.PsiConditionalExpression;

public class NumTernaryMethodCalculator extends NumSimpleElementCalculator {
    @Override
    public JavaRecursiveElementVisitor createVisitor() {
        return new NumTernaryMethodCalculator.Visitor();
    }

    private class Visitor extends NumSimpleElementCalculator.Visitor {
        @Override
        public void visitConditionalExpression(PsiConditionalExpression expression) {
            super.visitConditionalExpression(expression);
            elementsCounter++;
        }
    }
}

File: ApemanPlugin/src/main/kotlin/apeman_core/features_extraction/calculators/method/NumTernaryMethodCalculator.kt
package apeman_core.features_extraction.calculators.method

import apeman_core.base_entities.ExtractionCandidate
import apeman_core.base_entities.FeatureType
import com.intellij.psi.PsiConditionalExpression

class NumTernaryMethodCalculator(candidates: List<ExtractionCandidate>
) : NumSimpleElementMethodCalculator(candidates, FeatureType.CON_CONDITIONAL) {

    override fun createVisitor(methodCandidates: List<ExtractionCandidate>) = Visitor(methodCandidates)

    inner class Visitor(methodCandidates: List<ExtractionCandidate>
    ) : NumSimpleElementMethodCalculator.Visitor(methodCandidates) {
        override fun visitConditionalExpression(expression: PsiConditionalExpression) {
            super.visitConditionalExpression(expression)
            elementsCounter++
        }
    }
}
